DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,126 @@
+
+## Contents
+* [Rationale](#rationale)
+* [Description](#description)
+    * [Public Functions, Types, and Modules](#public-functions-types-and-modules)
+    * [Language Features](#language-features)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Rationale
+
+There is general disagreement on the best and/or accepted way to remove public
+features. Each deprecation ends up being handled with slight variations depending on
+who's writing the pull request. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, so as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out a new procedure for deprecating a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily determine if upgrading will break their
+code.
+
+Users must be given at least 10 non-patch releases before the deprecated features
+are removed. More releases can be given if the removed code is commonly used.
+There are two cases where the deprecation period is allowed to be shorter:
+
+1. The code or feature is notably dangerous or unsafe, and users need to remove
+it from their code as soon as possible.
+2. The existence of the current code precludes its own fix or the fix of an equally
+important issue.
+
+Shortening the deprecation period should be done with caution to avoid giving D
+an image of instability.
+
+At the time of the pull request for deprecation, all code in Phobos, Druntime,
+and DMD must be updated to remove use of the effected code. Maintainers of any
+projects which are tested on the Project Tester and are broken by the deprecation
+should be notified.
+
+Both at the time of deprecation and removal, a changelog entry must be made. This
+changelog entry should have a short motivation for the deprecation (or removal)
+and should describe which steps can be taken by the user to upgrade their codebase.
+
+In order to facilitate on schedule deprecations, a comment of the format
+`@@@DEPRECATED_[version]@@@` should be added to the top of the code to be removed/disabled.
+This comment allows code to be easily searched before every release to
+find all planned deprecations.
+
+### Public Functions, Variables, Types, and Modules
+
+All removals or changes to protection attributes of public functions,
+variables, types, and modules must be accompanied with a deprecation period.
+
+The symbol(s) must be marked using the `deprecated` keyword with a message containing
+the planned removal period. A reference to more information should also be added. E.g.
+"See the 2.080 changelog for more details" or "See the function documentation for more
+details". The documentation of the symbol(s) must be updated noting the
+deprecation and removal plan. The documentation should contain information to help
+the users of the symbol(s) transition their code away from the symbol(s).
```

Comment:
FYI: I finally got around proposing a `DEPRECATED` ddoc macro for consistent and prettier formatting of of deprecation messages in the spec + making it easier to grep for still documented and deprecated symbols.

https://github.com/dlang/dlang.org/pull/2308

Not sure if it's too late to incorporate this in the DIP.